### ISSUE TITLE
Failed to execute 'send' on 'WebSocket': Still in CONNECTING state.(…)

### ISSUE THREAD

**Issue by tresf** [qzind/tray#46]
Steps to reproduce:

1. Close QZ Tray if open
1. Open `sample.html` with a web browser
1. Launch QZ Tray
1. Quickly click "Connect" over and over until the page gets stuck
1. If you can't reproduce, try again until it happens.  About 1 out of 5 will do it for me if I click-jack enough.

Note, this is not addressed by https://github.com/qzind/tray/commit/e01235d9ee2e6ae14808b344a992e25049450142

```
qz-tray.js:41 DOMException: Failed to execute 'send' on 'WebSocket': Still in CONNECTING state.(…)
```

Web Browser's logs:

```
WebSocket connection to 'wss://localhost:8181/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
qz-tray.js:86 WebSocket connection to 'wss://localhost:8282/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
qz-tray.js:86 WebSocket connection to 'wss://localhost:8383/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
qz-tray.js:86 WebSocket connection to 'wss://localhost:8484/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
(index):1397 Error: Unable to establish connection with QZ(…)displayError @ (index):1397
qz-tray.js:86 WebSocket connection to 'wss://localhost:8181/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
qz-tray.js:86 WebSocket connection to 'wss://localhost:8282/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
qz-tray.js:86 WebSocket connection to 'wss://localhost:8383/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
qz-tray.js:86 WebSocket connection to 'wss://localhost:8484/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
(index):1397 Error: Unable to establish connection with QZ(…)displayError @ (index):1397
qz-tray.js:86 WebSocket connection to 'wss://localhost:8181/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
qz-tray.js:86 WebSocket connection to 'wss://localhost:8282/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
qz-tray.js:86 WebSocket connection to 'wss://localhost:8383/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
qz-tray.js:86 WebSocket connection to 'wss://localhost:8181/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
qz-tray.js:86 WebSocket connection to 'wss://localhost:8484/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
(index):1397 Error: Unable to establish connection with QZ(…)displayError @ (index):1397
qz-tray.js:86 WebSocket connection to 'wss://localhost:8282/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
qz-tray.js:86 WebSocket connection to 'wss://localhost:8181/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
qz-tray.js:86 WebSocket connection to 'wss://localhost:8383/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
2qz-tray.js:86 WebSocket connection to 'wss://localhost:8181/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
qz-tray.js:86 WebSocket connection to 'wss://localhost:8282/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
qz-tray.js:86 WebSocket connection to 'wss://localhost:8484/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
(index):1397 Error: Unable to establish connection with QZ(…)displayError @ (index):1397
qz-tray.js:37 Established connection with QZ Tray on wss://localhost:8181
qz-tray.js:41 DOMException: Failed to execute 'send' on 'WebSocket': Still in CONNECTING state.(…)_qz.log.error @ qz-tray.js:41
(index):1397 DOMException: Failed to execute 'send' on 'WebSocket': Still in CONNECTING state.(…)displayError @ (index):1397
qz-tray.js:37 Established connection with QZ Tray on wss://localhost:8181
qz-tray.js:41 DOMException: Failed to execute 'send' on 'WebSocket': Still in CONNECTING state.(…)_qz.log.error @ qz-tray.js:41
(index):1397 DOMException: Failed to execute 'send' on 'WebSocket': Still in CONNECTING state.(…)displayError @ (index):1397
qz-tray.js:37 Established connection with QZ Tray on wss://localhost:8181
qz-tray.js:41 DOMException: Failed to execute 'send' on 'WebSocket': Still in CONNECTING state.(…)_qz.log.error @ qz-tray.js:41
(index):1397 DOMException: Failed to execute 'send' on 'WebSocket': Still in CONNECTING state.(…)displayError @ (index):1397
qz-tray.js:37 Established connection with QZ Tray on wss://localhost:8181
qz-tray.js:41 DOMException: Failed to execute 'send' on 'WebSocket': Still in CONNECTING state.(…)_qz.log.error @ qz-tray.js:41
(index):1397 DOMException: Failed to execute 'send' on 'WebSocket': Still in CONNECTING state.(…)displayError @ (index):1397
qz-tray.js:37 Established connection with QZ Tray on wss://localhost:8181
qz-tray.js:41 DOMException: Failed to execute 'send' on 'WebSocket': Still in CONNECTING state.(…)_qz.log.error @ qz-tray.js:41
(index):1397 DOMException: Failed to execute 'send' on 'WebSocket': Still in CONNECTING state.(…)displayError @ (index):1397
qz-tray.js:37 Established connection with QZ Tray on wss://localhost:8181
qz-tray.js:41 DOMException: Failed to execute 'send' on 'WebSocket': Still in CONNECTING state.(…)_qz.log.error @ qz-tray.js:41
(index):1397 DOMException: Failed to execute 'send' on 'WebSocket': Still in CONNECTING state.(…)displayError @ (index):1397
qz-tray.js:37 Established connection with QZ Tray on wss://localhost:8181
qz-tray.js:41 DOMException: Failed to execute 'send' on 'WebSocket': Still in CONNECTING state.(…)_qz.log.error @ qz-tray.js:41
(index):1397 DOMException: Failed to execute 'send' on 'WebSocket': Still in CONNECTING state.(…)displayError @ (index):1397
2qz-tray.js:86 WebSocket connection to 'wss://localhost:8282/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
2qz-tray.js:86 WebSocket connection to 'wss://localhost:8383/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
qz-tray.js:37 Established connection with QZ Tray on wss://localhost:8181
qz-tray.js:41 DOMException: Failed to execute 'send' on 'WebSocket': Still in CONNECTING state.(…)_qz.log.error @ qz-tray.js:41
(index):1397 DOMException: Failed to execute 'send' on 'WebSocket': Still in CONNECTING state.(…)displayError @ (index):1397
qz-tray.js:86 WebSocket connection to 'wss://localhost:8484/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
(index):1397 Error: Unable to establish connection with QZ(…)displayError @ (index):1397
qz-tray.js:86 WebSocket connection to 'wss://localhost:8383/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
qz-tray.js:37 Established connection with QZ Tray on wss://localhost:8181
qz-tray.js:41 DOMException: Failed to execute 'send' on 'WebSocket': Still in CONNECTING state.(…)_qz.log.error @ qz-tray.js:41
(index):1397 DOMException: Failed to execute 'send' on 'WebSocket': Still in CONNECTING state.(…)displayError @ (index):1397
qz-tray.js:86 WebSocket connection to 'wss://localhost:8484/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
(index):1397 Connection to QZ Tray was closeddisplayError @ (index):1397
qz-tray.js:37 Closed connection with QZ Tray
(index):1332 CloseEvent
qz-tray.js:86 WebSocket connection to 'wss://localhost:8484/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
(index):1397 Connection to QZ Tray was closeddisplayError @ (index):1397
qz-tray.js:37 Closed connection with QZ Tray
qz-tray.js:153 Uncaught TypeError: Cannot read property 'promise' of null
qz-tray.js:37 Established connection with QZ Tray on wss://localhost:8181
qz-tray.js:145 Uncaught TypeError: Cannot set property 'established' of null
qz-tray.js:37 Established connection with QZ Tray on wss://localhost:8181
qz-tray.js:145 Uncaught TypeError: Cannot set property 'established' of null
qz-tray.js:37 Established connection with QZ Tray on wss://localhost:8181
qz-tray.js:145 Uncaught TypeError: Cannot set property 'established' of null
```

QZ Tray's logs:

```
[INFO] 2016-04-30 20:56:08,509 @ qz.auth.CRL:?
	Successfully loaded 1 CRL entries from https://crl.qz.io
[INFO] 2016-04-30 20:56:09,272 @ qz.ws.PrintSocketClient:?
	Connection opened from /0:0:0:0:0:0:0:1:61934 on socket port 8181
[INFO] 2016-04-30 20:56:09,492 @ qz.ws.PrintSocketClient:?
	Connection opened from /0:0:0:0:0:0:0:1:61939 on socket port 8181
[INFO] 2016-04-30 20:56:09,747 @ qz.ws.PrintSocketClient:?
	Connection opened from /0:0:0:0:0:0:0:1:61940 on socket port 8181
[INFO] 2016-04-30 20:56:09,879 @ qz.ws.PrintSocketClient:?
	Connection opened from /0:0:0:0:0:0:0:1:61941 on socket port 8181
[INFO] 2016-04-30 20:56:10,085 @ qz.ws.PrintSocketClient:?
	Connection opened from /0:0:0:0:0:0:0:1:61942 on socket port 8181
[INFO] 2016-04-30 20:56:10,669 @ qz.ws.PrintSocketClient:?
	Connection opened from /0:0:0:0:0:0:0:1:61943 on socket port 8181
[INFO] 2016-04-30 20:56:10,754 @ qz.ws.PrintSocketClient:?
	Connection opened from /0:0:0:0:0:0:0:1:61944 on socket port 8181
[INFO] 2016-04-30 20:56:11,393 @ qz.ws.PrintSocketClient:?
	Connection opened from /0:0:0:0:0:0:0:1:61953 on socket port 8181
[INFO] 2016-04-30 20:56:11,748 @ qz.ws.PrintSocketClient:?
	Connection opened from /0:0:0:0:0:0:0:1:61958 on socket port 8181
[INFO] 2016-04-30 20:56:12,274 @ qz.ws.PrintSocketClient:?
	Connection opened from /0:0:0:0:0:0:0:1:61963 on socket port 8181
[INFO] 2016-04-30 20:56:12,473 @ qz.ws.PrintSocketClient:?
	Connection opened from /0:0:0:0:0:0:0:1:61964 on socket port 8181
[INFO] 2016-04-30 20:56:13,451 @ qz.ws.PrintSocketClient:?
	Connection opened from /0:0:0:0:0:0:0:1:61965 on socket port 8181
```

**Comment by bberenz** [comment]
> Note, this is not addressed by e01235d

I wouldn't expect it to be, since this is all browser side
Handled now by 9e8ab20ca99098d8aa40e527c2fdcf8ae6fa477a

Side note, in Firefox this error shows up as:
`An attempt was made to use an object that is not, or is no longer, usable`

**Comment by bberenz** [comment]
Let's include 714283bc3aa567ac272778e0cbbfcfe7f0088881 on that fix too..
